backend-service/tsconfig.json

```diff
@@ -0,0 +1,111 @@
+{
```

Comment:
Sadly `tsc --init`'s default tsconfig is pretty rubbish. It doesn't prompt you to find what type of project your building (backend, react, npm library, etc) so it doesn't actually give a good default. It's also very old old and it's hard to know what's going on because of all the comments. 

[This cheatsheet](https://www.totaltypescript.com/tsconfig-cheat-sheet) is a much better starting point. 

I would specifically recommend this: 

```json
{
  "compilerOptions": {
    /* Base Options: */
    "esModuleInterop": true,
    "skipLibCheck": true,
    "target": "es2022",
    "allowJs": true,
    "resolveJsonModule": true,
    "moduleDetection": "force",
    "isolatedModules": true,
    "verbatimModuleSyntax": true,

    /* Strictness */
    "strict": true,
    "noUncheckedIndexedAccess": true,
    "noImplicitOverride": true,

    /* If transpiling with TypeScript: */
    "module": "NodeNext",
    "outDir": "dist",
    "sourceMap": true,


    /* If your code doesn't run in the DOM: */
    "lib": ["es2022"]
  }
}
```

You'll have to add `.js` when you import files (even ts files) with this config which seems a bit counter intuitive but TypeScript is basically just a VSCode plugin so it does make sense and it's required for some of the more strict type checking.